JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt

```diff
@@ -121,103 +139,160 @@ fun HomeScreen(
-        // don't restart the effect and use the latest lambda values.
-        val onRefreshPostsState by rememberUpdatedState(onRefreshPosts)
-        val onErrorDismissState by rememberUpdatedState(onErrorDismiss)
+                // If onRefreshPosts or onErrorDismiss change while the LaunchedEffect is running,
+                // don't restart the effect and use the latest lambda values.
+                val onRefreshPostsState by rememberUpdatedState(onRefreshPosts)
+                val onErrorDismissState by rememberUpdatedState(onErrorDismiss)
 
-        // Effect running in a coroutine that displays the Snackbar on the screen
-        // If there's a change to errorMessageText, retryMessageText or scaffoldState,
-        // the previous effect will be cancelled and a new one will start with the new values
-        LaunchedEffect(errorMessageText, retryMessageText, scaffoldState) {
-            val snackbarResult = scaffoldState.snackbarHostState.showSnackbar(
-                message = errorMessageText,
-                actionLabel = retryMessageText
-            )
-            if (snackbarResult == SnackbarResult.ActionPerformed) {
-                onRefreshPostsState()
+                // Effect running in a coroutine that displays the Snackbar on the screen
+                // If there's a change to errorMessageText, retryMessageText or scaffoldState,
+                // the previous effect will be cancelled and a new one will start with the new values
+                LaunchedEffect(errorMessageText, retryMessageText, scaffoldState) {
+                    val snackbarResult = scaffoldState.snackbarHostState.showSnackbar(
+                        message = errorMessageText,
+                        actionLabel = retryMessageText
+                    )
+                    if (snackbarResult == SnackbarResult.ActionPerformed) {
+                        onRefreshPostsState()
+                    }
+                    // Once the message is displayed and dismissed, notify the ViewModel
+                    onErrorDismissState(errorMessage.id)
+                }
+            }
+        } else {
+            val selectedPost = uiState.posts.find { it.id == uiState.selectedPostId }
+
+            if (selectedPost != null) {
+                val lazyListState = detailLazyListStates.getValue(selectedPost.id)
+
+                ArticleScreen(
+                    post = selectedPost,
+                    onBack = onInteractWithList,
+                    isFavorite = uiState.favorites.contains(selectedPost.id),
+                    onToggleFavorite = {
+                        onToggleFavorite(selectedPost.id)
+                    },
+                    lazyListState = lazyListState,
+                )
+
+                // If we are just showing the detail, have a back press switch to the list.
+                if (!useListDetail) {
+                    BackHandler {
+                        onInteractWithList()
```

Comment:
Can we make Navigation do this for us?

Reply:
Technically yes, _but_ that would require performing a navigate upon switching from the list/detail to the detail screen, which we want to avoid for a couple of reasons:
- Performing navigation in a side-effect results in some jank, since it happens after composition
- We'd also need to do another navigate upon switching back from detail to the list/detail screen to preserve reversability